transforms/dct.go

```diff
@@ -75,25 +75,49 @@ func DCT2D(input [][]float64, w int, h int) [][]float64 {
 }
 
 // DCT2DFast64 function returns a result of DCT2D by using the seperable property.
-// Fast uses static DCT tables for improved performance.
-func DCT2DFast64(input *[]float64) {
-	if len(*input) != 4096 {
-		panic("incorrect input size")
+// Fast uses static DCT tables for improved performance. Returns flattened pixels.
+func DCT2DFast64(input *[]float64) (flattens [64]float64) {
+	if len(*input) != 64*64 {
+		panic("incorrect input size, wanted 64x64.")
 	}
 
 	for i := 0; i < 64; i++ { // height
-		DCT1DFast64((*input)[i*64 : (i*64)+64])
-		//forwardTransformFast((*input)[i*64:(i*64)+64], temp[:], 64)
+		forwardDCT64((*input)[i*64 : (i*64)+64])
 	}
 
-	for i := 0; i < 64; i++ { // width
-		row := [64]float64{}
+	var row [64]float64
+	for i := 0; i < 8; i++ { // width
 		for j := 0; j < 64; j++ {
-			row[j] = (*input)[i+((j)*64)]
+			row[j] = (*input)[64*j+i]
 		}
-		DCT1DFast64(row[:])
-		for j := 0; j < len(row); j++ {
-			(*input)[i+(j*64)] = row[j]
+		forwardDCT64(row[:])
+		for j := 0; j < 8; j++ {
+			flattens[8*j+i] = row[j]
 		}
 	}
+	return
+}
+
+// DCT2DFast256 function returns a result of DCT2D by using the seperable property.
+// DCT type II, unscaled. Algorithm by Byeong Gi Lee, 1984.
+// Fast uses static DCT tables for improved performance. Returns flattened pixels.
+func DCT2DFast256(input *[]float64) (flattens [256]float64) {
```

Comment:
Can you explain where DCT2DFast256 will be used for?
I can not read the use case of this function in this PR, it will be the fast path for ExtPerceptionHash?

Reply:
Yes that was the thought process. For images that are 256x256 pixels. There can be larger ExtPerceptionHash but I believe these would be less common.

Reply:
@evanoberholster 

Would you like to integrate this function to ExtPerceptionHash through this PR?

Reply:
I left it incomplete as I am unsure how you would like to integrate a 256x256 Hash into ExtPerceptionHash. If someone wanted to create a 512x512 hash then there would need to be an additional ExtPerceptionHash function.

I don't believe that 512x512 is useful, but someone may have this use case. I would appreciate your recommendation.

Reply:
Hmm sorry
One last question: Why are you trying to add DCT2DFast256 through this PR?

Reply:
No. it would need further work. You may remove. Thanks